web3/providers/websocket/websocket.py

```diff
@@ -2,134 +2,243 @@
 import json
 import logging
 import os
-from threading import (
-    Thread,
-)
-from types import (
-    TracebackType,
-)
 from typing import (
     Any,
+    Dict,
     Optional,
-    Type,
     Union,
 )
 
 from eth_typing import (
     URI,
 )
+from toolz import (
+    merge,
+)
 from websockets.client import (
     connect,
 )
-from websockets.legacy.client import (
-    WebSocketClientProtocol,
+from websockets.exceptions import (
+    WebSocketException,
```

Comment:
consistency nit - All the Providers are `Websocket...`, but the Exception (and a couple other places I found in search) use `WebSocket...` Probably good to make them all `Websocket`, yah?